drivers/include/mcp47xx.h

```diff
@@ -0,0 +1,347 @@
+ * used by device.
+ */
+typedef enum {
+    MCP4706,    /**< 1 channel 8-bit DAC */
+    MCP4716,    /**< 1 channel 10-bit DAC */
+    MCP4725,    /**< 1 channel 12-bit DAC */
+    MCP4726,    /**< 1 channel 12-bit DAC */
+    MCP4728,    /**< 4 channel 12-bit DAC */
+} mcp47xx_variant_t;
+
+/**
+ * @brief   MCP47xx gain configuration type
+ *
+ * @note Gains are not supported by MCP4725.
+ */
+typedef enum {
+    MCP47XX_GAIN_1X = 0,  /**< Gain is 1.0, supported by all MCP47xx variants */
+    MCP47XX_GAIN_2X = 1,  /**< Gain is 2.0, not supported by MCP4725 */
+} mcp47xx_gain_t;
+
+/**
+ * @brief   MCP47xx V_REF configuration type
+ *
+ * @note Different MCP47xx variants allow different V_REF configurations
+ */
+typedef enum {
+    MCP47XX_VREF_VDD = 0,   /**< V_REF = V_DD, supported by all MCP47xx */
+    MCP47XX_VREF_INT = 1,   /**< V_REF = internal (2.048 V), MCP4728 only  */
+    MCP47XX_VREF_PIN = 2,   /**< V_REF = VREF pin not buffered, MCP47x6 only */
+    MCP47XX_VREF_BUF = 3,   /**< V_REF = VREF pin buffered, MCP47x6 only */
+} mcp47xx_vref_t;
+
+/**
+ * @brief   MCP47xx Power-down mode selection type
+ *
+ * Defines the possible power-down modes used for MCP47xx device configuration.
+ * The mode is used by function #mcp47xx_dac_poweroff to set the DAC into the
+ * configured power-down mode.
+ *
+ * @note #MCP47XX_NORMAL cannot be configured as power-down mode.
+ */
+typedef enum {
+    MCP47XX_NORMAL = 0,     /**< Normal mode */
+    MCP47XX_PD_SMALL = 1,   /**< Power down, small resistor 1 kOhm */
+    MCP47XX_PD_MEDIUM = 2,  /**< Power down, medium resistor,
+                                 125 kOhm for MCP47x6, 100 kOhm otherwise */
+    MCP47XX_PD_LARGE = 3,   /**< Power down, large resistor,
+                                 640 kOhm for MCP47x6, 125 kOhm otherwise */
+} mcp47xx_pd_mode_t;
+
+/**
+ * @brief   MCP47xx device configuration parameters
+ */
+typedef struct {
+
+    i2c_t    dev;       /**< I2C device */
+    uint16_t addr;      /**< I2C slave address MCP47XX_BASE_ADDR + [0...7] */
+
+    mcp47xx_variant_t   variant;    /**< used variant of MCP47xx */
+    mcp47xx_gain_t      gain;       /**< Gain selection */
+    mcp47xx_vref_t      vref;       /**< Voltage reference selection */
+    mcp47xx_pd_mode_t   pd_mode;    /**< Power-down mode selection */
+
+} mcp47xx_params_t;
+
+/**
+ * @brief   MCP47xx device data structure type
+ */
+typedef struct {
+    mcp47xx_params_t params;              /**< device configuration parameters */
+    uint16_t values[MCP47XX_CHN_NUM_MAX]; /**< contains the last values set */
+} mcp47xx_t;
+
+#if IS_USED(MODULE_SAUL) || DOXYGEN
+/**
+ * @brief   MCP47xx configuration structure for mapping DAC channels to SAUL
+ */
+typedef struct {
+    const char *name;   /**< name of the MCP47xx device */
+    unsigned int dev;   /**< index of the MCP47xx device */
+    dac_t channel;      /**< channel of the MCP47xx device */
```

Comment:
`dac_t` ?

Reply:
Yes, it is the same as for the `perhiph/dac`. The goal was to keep it as compatible as possible with `periph/dac` and to handle channels in same way when we intended to introduced the DAC extension API with PR #10512.

Reply:
But `dac_t` would refer to a unique DAC device, not a channel of a DAC.
Just using `uint8_t` here would be less confusing IMHO (there also is no `saul_dac_params_t` yet)

Reply:
No, `dac_t` refers a line, that device + channel. It is also used in `periph/dac` to [set a certain channel of a device](https://doc.riot-os.org/group__drivers__periph__dac.html#ga179ae55de04d356cd6fbb63e9e3f51f6). In this sense, it also makes sense to use it here.

Reply:
When we started to introduce the extension API, we used exactly the same types for extenders to refer devices and channels.

Reply:
Unfortunatly, we had not much luck with the extension interfaces as suggested in PR #10512 thil now.

Reply:
However, we can of course also use `uint8_t`, if there is an extension API at some point in future, the type `dac_t` can still be mapped to `dev` + `chn`.

Reply:
We can still change it later if we go forward with the extension API. :wink: 
To an unsuspecting user like me the `dac_t` type here just adds confusion.